driver-core/src/main/java/com/datastax/driver/core/SimpleStatement.java

```diff
@@ -117,7 +114,24 @@ public String getQueryString() {
 
     @Override
     public ByteBuffer[] getValues() {
```

Comment:
As a side note, 2.1 and 2.2 branches (including custom codecs) both already store objects, not byte buffers, and serialization takes place every time this method is called.